runtime/common/orderedmap/orderedmap.go

```diff
@@ -229,6 +232,35 @@ func (om *OrderedMap[K, V]) ForAllKeys(predicate func(key K) bool) bool {
 	return true
 }
 
+// MapKeys returns a new ordered map whose keys are mapped according to
+// the provided mapping function between
+func MapKeys[T OrderedMap[K, V], K comparable, V any, H comparable](om *OrderedMap[K, V], f func(K) H) *OrderedMap[H, V] {
+	mapped := New[OrderedMap[H, V]](om.Len())
+
+	om.Foreach(func(key K, value V) {
+		mapped.Set(f(key), value)
+	})
+
+	return mapped
+}
+
+// SortByKeys returns a new ordered map whose insertion order is sorted according to
+// the provided comparison function between keys
+func (om *OrderedMap[K, V]) SortByKey(compare func(K, K) bool) *OrderedMap[K, V] {
+	sorted := New[OrderedMap[K, V]](om.Len())
+	keys := maps.Keys(om.pairs)
```

Comment:
Need to ignore the linting errors here by adding `//nolint:forbidigo` and a comment explaining why it's OK, i.e. the result is sorted afterwards